src/ixdat/readers/zilien.py

```diff
@@ -135,25 +148,27 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
 
         # Extract timestamp from filename on form: 2021-04-20 11_16_18 Measurement name
         file_stem = self._path_to_file.stem  # Part of filename before the extension
-        timestamp = timestamp_string_to_tstamp(
+        self._timestamp = timestamp_string_to_tstamp(
```

Comment:
Good point. I'm sorry I missed that. We **do** actually add the timestamp as a metadata item now. It was always supposed to be there and it was a bug that it was missed in the first place. Point is that now, if the timestamp is in metadata we should use that, if not, then we fall back to using the one from the file name.

Reply:
Now, it will check the unix timestamp in the file metadata first. Only when it is not there it will try to parse it from the filename.

Reply:
Although the difference between these two is about 3 seconds. I am finding out why in our codebase now,

Reply:
As in, it's 3 s slower to read the tstamp from inside the file than from the title??

Reply:
No no, it is due to a synchronization of the file creation and a core initialization reasons. But now, the Zilien files will _always_ include the (correct) timestamp and it will _always_ be read from the file. So I am up for leaving the (wrong) timestamp in the filename purely for informative and sorting reasons. The time difference between the two timestamps depends on how fast a user's computer is.
Unfortunately it is not easily possible to put the correct timestamp into the filename.

Reply:
Great!
Understood - that's the time it took Zilien to make the .tsv from the data in the tmp folder, then, I guess?

Reply:
No. Let me try and clear up a few things, because as it actually turns out the implications here are a little bigger than just an annoyance.

The timestamp in the filename is the time at which the folders are created at measurement start and is only used to make sure that every started measurement will get a unique folder/file-name, even with identical project and measurement names. This is different by a few seconds from the measurement timestamp, which is the timestamp designated as measurement time 0.0 (and what ixdat understands as the timestamp) after all the software components reports ready for measurement.

The annoying conclusion to this is that the timestamp in the filename **should not be used to synchronisation**, since that will give you a few seconds offset. The problem is, that since I missed this difference (I had forgotten that it was not the measurement timestamp in the filename), then the way that ixdat has done it so far has actually produced the wrong result and therefore measurements whose sync is of by a few seconds!!

For the new file version, all will be good, because we have taken care of the sync. But for the existing data sets I think what should happen, in order to make sure that the sync is good, is that when the separate MS part in a Zilien file gets combined with a EC-Lab measuement, that the Zilien measurement knows about, then it should adjust the timestamp of the EC-Measurement to that of the Zilien measurement + the offset in the metadata. This might be accomplished by some clever overloading the `__add__` but I will need to give it some more thought.

Reply:
Even with this comment. I consider what I mentioned above a bug and will open an issue. I think it is all still correct for this PR so I will leave this comment resolved.